docs/micro-service-architecture.md

```diff
@@ -0,0 +1,248 @@
+# Architecture document
+The goal of this document is to provide a clear overview of our microservice infrastructure, explain our choices and provide a clear OpenAPI definition.
+
+## Technologies
+
+#### `Kubernetes`
+
+Our production architecture leverages Kubernetes for container orchestration with Docker as the containerization platform. 
+This powerful combination provides scalability, flexibility, and efficient management of containerized applications.
+
+#### `Docker`
+
+We mianly use docker for development with many docker-composes to repliate the micro services and their databases but as said before we 
+also use Docker as the containerization platform in our kubernetes environment
+
+#### `NATS`
+
+We chose this solution instead of Kafka for it's lightweight design, low latency and overall simplicity. It also has a built-in mechanism 
+for doing synchronous request-reply messaging, which will be very useful for communication between the API gateway and the microservices. 
+
+#### `Kotlin`
+
+We chose kotlin as our main backend language because we wanted to try something else than java
+
+#### `Spring boot`
+
+Our architecture utilizes Spring Boot for building and deploying containerized Kotlin applications. 
+Spring Boot simplifies the development process, enhances productivity, and provides a solid foundation for microservices. 
+It was also one of the mandatory elements of the app
+
+#### `React native`
+
+React native was kind of mandatory in this project, we could have been using flutter but since we never tried any of them React native seemed like a 
+great opportunity to complete our React dev stack.
+
+#### `Expo`
+
+Our mobile application architecture is built with the Expo framework.
+Expo simplifies the development of cross-platform mobile apps, providing tools and services that enhance the developer experience.
+It was recommanded for the development of this project
+
+#### `Github`
+
+We chose github as our primary version controll and collaboration platform since we both use it on a daily basis.
+
+#### `Github Action`
+
+Github Action seemed like an obvious choise when using Github since it's great tool to create and manage pipelines for both Continuous Integration and Continuous Deployment
+
+
+#### `Keycloak`
+[Keycloak](https://www.keycloak.org/) is an open source identity and access management solution. 
+We decided to use keycloak instead of another service like to acquire more experience on it since it is becoming one of the market leader.
```

Comment:
```suggestion
### Keycloak

[Keycloak](https://www.keycloak.org/) is an open source identity and access management solution. 

It is used to provide a secure authentication and authorization experience for the users. 

We decided to use Keycloak instead of another service like Zitadel or Auth0 to acquire more experience on it since it is one of the most popular identity providers.
```